sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -1658,4 +1658,49 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
     val df = spark.createDataFrame(spark.sparkContext.makeRDD(rows), schema)
     assert(df.filter($"array1" === $"array2").count() == 1)
   }
+
+  test("SPARK-18207: Compute hash for wider table") {
+    import org.apache.spark.sql.types.{StructType, StringType}
```

Comment:
I do agree with @viirya the current test is way to indirect and can actually be broken silently if the planner choses a sort based aggregate over a hash based aggregate. It would be very nice to have a direct test on the hash function.